File: python/merge/_Merger_png_pdf.py
from Aspose.Imaging import Rectangle
from Aspose.Imaging import Image
from aspose.pdf import SaveFormat
from aspose.pdf import PageSize
from aspose.pdf import Document
import clr

aspose_pdf = clr.addReference("../../lib/Aspose.PDF.dll")
aspose_imaging = clr.addReference("../../lib/Aspose.Imaging.dll")

class png_to_pdf(object):
    def __init__(self, licence_path):
        self.dataDir = "../../TestData"
        if licence_path:
            self.licence_path = licence_path
            self.aspose_license = License()
            self.aspose_license.SetLicense(self.licence_path)

    def exec():
        path_source1 = "../../TestData/test.png"
        path_source2 = "../../TestData/Second/test.png"

        # create empty pdf document
        doc = Document

        # set less memory usage with unload instead of fast performance
        doc.enable_object_unload = True

        # make list of files with images to merge
        images = [path_source1, path_source2]

        for fs in images:
            # add new page to pdf
            page = doc.pages.add()

            # setup page size to be A4
            page.SetPageSize(PageSize.a4.width, PageSize.a4.height)

            # load image from stream, it supports a lot of formats
            image = Image.Load(fs)
            # read image dimensions to pdf page rectangle
            rect = Rectangle(0, 0, image.width - 1, image.height - 1)

            # add image to new pdf page
            page.addImage(fs, rect)

        # save result pdf to file
        doc.save("test.pdf", SaveFormat.Pdf)

File: cpp/merge/_Merger_png_pdf.cpp
#include "Aspose.PDF.Cpp/Document.h"
#include "Aspose.PDF.Cpp/Page.h"
#include "Aspose.PDF.Cpp/PageCollection.h"
#include "Aspose.PDF.Cpp/PageSize.h"
#include "Aspose.PDF.Cpp/Generator/Image.h"
#include "Aspose.PDF.Cpp/Rectangle.h"
#include "Aspose.PDF.Cpp/SaveFormat.h"

using namespace System;
using namespace Aspose::Pdf;

void png_to_pdf()
{
    String pathSource1 = u"../../TestData/test.png";
    String pathSource2 = u"../../TestData/Second/test.png";

    // create empty pdf document
    auto outputDoc = MakeObject<Document>();

    // set less memory usage with unload instead of fast performance
    doc->set_EnableObjectUnload(true);

    // make list of files with images to merge
    String images[] = {pathSource1, pathSource2};

    for (int i = 1; i < sizeof(images); i++)
    {
        auto fs = images[i];
        // add new page to pdf
        auto page = document->get_Pages()->CopyPage();

        // setup page size to be A4
        page->SetPageSize(PageSize::get_A4()->get_Width(), PageSize::get_A4()->get_Height());

        //todo: calc image size

        com.aspose.pdf.Rectangle rect;
        // load image from stream, it supports a lot of formats
        com.aspose.imaging.Image image = com.aspose.imaging.Image.load(fs);
        // read image dimensions to pdf page rectangle
        rect = new com.aspose.pdf.Rectangle(0, 0, image.getWidth() - 1, image.getHeight() - 1);

        // add image to new pdf page
        page->AddImage(fs, rect);
    }

    // save result pdf to file
    doc->Save(u"test.pdf", SaveFormat::Pdf);
}
